File: backend/src/main/java/de/grimsi/gameyfin/entities/Library.java
package de.grimsi.gameyfin.entities;

import lombok.*;
import org.hibernate.Hibernate;

import javax.persistence.*;
import java.util.List;
import java.util.Objects;

@Entity
@Builder
@Getter
@Setter
@ToString
@AllArgsConstructor
@RequiredArgsConstructor
public class Library {

    @Id
    private String path;

    @ManyToMany(cascade = CascadeType.MERGE)
    @ToString.Exclude
    private List<Platform> platforms;

    @Override
    public boolean equals(Object o) {
        if (this == o) return true;
        if (o == null || Hibernate.getClass(this) != Hibernate.getClass(o)) return false;
        Library library = (Library) o;
        return path != null && Objects.equals(path, library.path);
    }

    @Override
    public int hashCode() {
        return getClass().hashCode();
    }
}

File: src/main/kotlin/de/grimsi/gameyfin/libraries/entities/Library.kt
package de.grimsi.gameyfin.libraries.entities

import jakarta.persistence.Entity
import jakarta.persistence.GeneratedValue
import jakarta.persistence.GenerationType
import jakarta.persistence.Id

@Entity
class Library(
    @Id
    @GeneratedValue(strategy = GenerationType.AUTO)
    var id: Long? = null,

    var path: String
)